en/appendices/4-0-upgrade-guide.rst

```diff
@@ -0,0 +1,61 @@
+4.0 Upgrade Guide
+#################
+
+Before attempting to upgrade to 4.0 first ensure you have upgraded to 3.8. Then
+enable deprecation warnings::
+
+    // in config/app.php
+    'Error' => [
+        'errorLevel' => E_ALL ^ E_USER_DEPRECATED,
+    ]
+
+Then, incrementally fix the deprecation warnings your application and its
+plugins emit.
+
+Use the Upgrade Tool
+====================
+
+Because CakePHP 4 adopts strict mode and uses more typehinting there are many
+backwards incompatible changes concerning method signatures, and file renames.
+To help expedite fixing these tedious changes there is an upgrade CLI tool::
+
```

Comment:
Grammar suggestions:

Because CakePHP 4 adopts strict mode and uses more typehinting, there

concerning method signatures and file renames.